### DISCUSSION TITLE
Programmatic scrolling with Vue

### DISCUSSION THREAD

**Discussion by binairo** [inokawa/virtua#513] [Q&A] [answered]
Hi, I'm trying to scroll the `VList` programmatically to a specific index. However, I cannot seem to get a reference to the `VList`

Basically trying to do this:

```vue
<template>
  <VList ref="scroller" ...>
    ...
  </Vlist>
</template>

<script>
...
import { VList } from 'virtua/vue';
export default {
  ...
  setup(props, ctx) {
    const scroller = ref(null);  // using Vue 3.4
    watch(scroller, ()=>{
      // this never happens. `scroller.value` is always `null`
    });
    const doSomeScrolling = () => {
      // FAILS: scroller.value is always null 
      scroller.value.scrollBy(100);
    }
  }
}
</script>
```

I'm at a loss, but maybe the `VList` is set up in such a way that this doesn't work?

Any help is greatly appreciated!

**Comment by inokawa** [comment]
Could you make a reproduction? It did not reproduce in my environment.

**Reply by binairo** [reply, marked as answer]
Thank you for having a look.

I had to return the `scroller` ref at the end of the `setup` function to make it work 🤦
